docs/internals/dump_format.rst

```diff
@@ -0,0 +1,135 @@
+Dump File Format
+================
+
+This description uses the same :ref:`conventions <ref_protocol_conventions>`
```

Comment:
We should preface this with a warning that the dump format isn't yet stable and might change before 1.0.

Reply:
Added a warning